terraform/deployments/ecr/main.tf

```diff
@@ -21,32 +25,28 @@ provider "aws" {
   }
 }
 
-# TODO: Get rid of this list and just give CI permission to create repos, e.g.
-# with https://github.com/byu-oit/github-action-create-ecr-repo-if-missing
```

Comment:
Did you consider giving CI permission to create ECR repos as needed? It'd be nice not to have to apply this terraform whenever we start a new project.

Reply:
Allowing the GHA to create an ECR repo would be nice. We'd still need a method to update and delete ECR repositories (e.g. we want to modify the [`image_scanning_configuration`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecr_repository#image_scanning_configuration), which would be better achieved with Terraform. Also note that further down this file the ARNs of repos are used to add policies to the IAM role, which would need to be broader (i.e. `arn::.../*` rather than `arn::.../content-store`). Finally, in the `tf/deploys/github` module, we already need to apply TF after a new project is added, in order to provide credentials to the GH Action.

I think a nice solution would be to have an observer component (e.g. a Concourse input) trigger Terraform applies whenever the set of repos with the tags "govuk,docker,archived:false,..." changes.